src/main/java/com/shaft/cli/FileActions.java

```diff
@@ -443,10 +435,13 @@ public static boolean zipFiles(String srcFolder, String destZipFile) {
         return result;
     }
 
+    @SuppressWarnings("UnusedReturnValue")
     public static File unpackArchive(URL url, String destinationFolderPath) {
         File targetDir = new File(destinationFolderPath);
         if (!targetDir.exists()) {
-            targetDir.mkdirs();
+            if (!targetDir.mkdirs()) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5739304)